youtube_dl/extractor/tube8.py

```diff
@@ -1,86 +1,260 @@
 from __future__ import unicode_literals
 
+import itertools
 import re
+from time import sleep
 
+from .common import InfoExtractor
 from ..utils import (
+    traverse_obj,
+    T,
+    url_or_none,
+    parse_iso8601,
+    get_element_by_class,
+    get_element_by_id,
     int_or_none,
-    str_to_int,
+    parse_qs,
+    urljoin,
 )
-from .keezmovies import KeezMoviesIE
 
 
-class Tube8IE(KeezMoviesIE):
-    _VALID_URL = r'https?://(?:www\.)?tube8\.com/(?:[^/]+/)+(?P<display_id>[^/]+)/(?P<id>\d+)'
+class Tube8IE(InfoExtractor):
+    _VALID_URL = r'https?:\/\/(?:www\.)?tube8\.com\/+porn-video+\/(?P<id>\d+)'
     _TESTS = [{
-        'url': 'http://www.tube8.com/teen/kasia-music-video/229795/',
-        'md5': '65e20c48e6abff62ed0c3965fff13a39',
+        'url': 'https://www.tube8.com/porn-video/189530841/',
+        'md5': '532408f59e89a32027d873af6289c85a',
         'info_dict': {
-            'id': '229795',
-            'display_id': 'kasia-music-video',
+            'id': '189530841',
             'ext': 'mp4',
-            'description': 'hot teen Kasia grinding',
-            'uploader': 'unknown',
-            'title': 'Kasia music video',
-            'age_limit': 18,
-            'duration': 230,
-            'categories': ['Teen'],
-            'tags': ['dancing'],
+            'title': 'Found dildo. She let it cum in her tight ass to keep the secret',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': 'MaryKrylova',
+            'timestamp': 1718961736,
+            'upload_date': '20240621',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_valid_url(url).group('id')
+        webpage = self._download_webpage(url, video_id)
+
+        playervars = self._search_json(
+            r'\bplayervars\s*:', webpage, 'playervars', video_id)
+
+        extra_info = self._search_json(
+            r'application/ld\+json[\"\']?>\s?', webpage, 'extra_info', video_id)
+
+        uploader = traverse_obj(extra_info, (
+            '@graph', lambda _, v: v.get('author'), 'author'))[0]
+
+        thumbnail = traverse_obj(extra_info, (
+            '@graph', lambda _, v: v.get('thumbnail'), 'thumbnail'))[0]
+
+        timestamp = parse_iso8601(traverse_obj(extra_info, (
+            '@graph', lambda _, v: v.get('datePublished'), 'datePublished'))[0])
```

Comment:
I would put the check for the age verify message at the top, after the webpage is downloaded. So I'm not in the USA and with my VPN, I cloud not verify the message to appear. So someone else must verify the checker.

Reply:
`duration` isn't one of the keys that the test harness proposes, but if you add `'duration': 0,` to the test case info-dict, it will complain:
```console
AssertionError: Invalid value for field duration, expected 0, got 686.0
```
Check the function `expect_info_dict()` in `test/helper.py` to see the list keys that get proposed if missing from  the test case:
```
                          ...
                          if value and key in ('id', 'title', 'description', 'uploader', 'upload_date', 'timestamp', 'uploader_id', 'location', 'age_limit'))
...
```